packages/core/echo/echo/src/internal/types/base.ts

```diff
@@ -0,0 +1,60 @@
+//
+// Copyright 2024 DXOS.org
+//
+
+import * as Schema from 'effect/Schema';
+import * as SchemaAST from 'effect/SchemaAST';
+
+import { type ObjectId } from '@dxos/keys';
+
+import { type ATTR_META, type ObjectMeta } from './meta';
+
+/**
+ * Marker interface for object with an `id`.
+ */
+export interface HasId {
+  readonly id: ObjectId;
+}
+
+/**
+ * Object that has an associated typename.
+ * The typename is retrievable using {@link getTypename}.
+ * The object can be used with {@link isInstanceOf} to check if it is an instance of a schema.
+ */
+export type HasTypename = {};
```

Comment:
is this still useful?

We don't need aspects anymore after simplifying object ontology -- all objects have:

- id
- kind: object, relation
- typename
- schema
- meta
- deletion marker